src/index.ts

```diff
@@ -1,59 +1,52 @@
+  contentStripping: /<\/?[^>]+(>|$)/g,
+  dataTransformer: undefined, // No transformation
+  searchComponentPath: 'vitepress-plugin-simple-search/SimpleSearch.vue',
+  searchOptions: {},
+  searchText: 'Search...',
+  placeholderText: 'Search. Use double quotes for stricter results.',
+  noResultsText: 'No results found.',
+  searchTimeText: '{time}ms for {count} results',
+});
 
 /**
  * Initialize the plugin, and pass additional configuration options.
  *
  * @export
- * @param {Options} options
+ * @param {UserOptions} options
  * @return {Plugin}
  */
-export function SimpleSearch(userOptions: Options): Plugin {
-    const options = setOptions(userOptions);
-
-    let config: any;
-    const virtualModuleId = 'virtual:simple-search';
-    const resolvedVirtualModuleId = '\0' + virtualModuleId;
+export function SimpleSearch(userOptions: Partial<Options> = {}): Plugin {
+  const options = { ...DEFAULT_OPTIONS, ...userOptions } as Options;
 
-    return {
-        name: 'simple-search',
-        enforce: 'pre',
-        configResolved(resolvedConfig) {
-            config = resolvedConfig;
-        },
-        config: () => ({
-            resolve: {
-                alias: { './VPNavBarSearch.vue': 'vitepress-plugin-simple-search/Search.vue' },
-            },
-        }),
-        resolveId(id) {
-            if (id === virtualModuleId) {
-                return resolvedVirtualModuleId;
-            }
+  return {
+    name: 'simple-search',
+    enforce: 'pre',
+    config: () => ({
+      resolve: {
+        alias: {
+          './VPNavBarSearch.vue': options.searchComponentPath,
         },
-        async load(this, id) {
-            if (id === resolvedVirtualModuleId) {
-                const filePathway = options.docsPath ? options.docsPath : config.root;
-                if (!fs.existsSync(filePathway)) {
-                    console.warn(`File Pathway: ${filePathway} does not exist. Search could not be built.`)
-                    console.warn(`Try 'process.cwd() + your folder'`)
-                    throw new Error(`Docs pathway could not be found.`);
-                }
+      },
+    }),
+    resolveId(id) {
+      if (id === 'virtual:simple-search') {
+        return '\0virtual:simple-search';
+      }
+    },
+    async load(this, id) {
+      if (id === '\0virtual:simple-search') {
+        const indexedData = await build(options);
 
-                const fileData = await buildDocumentation(config.root, userOptions.preambleTransformer);
-                const javaScript: string =
-                    `const regexForContentStripping = ${options.regexForContentStripping}` +
-                    ';\n' +
-                    'const data =' +
-                    JSON.stringify(fileData) +
-                    ';\n' +
-                    'export default { data, regexForContentStripping };';
-
-                return javaScript;
-            }
-        },
-    };
+        return `export default ${JSON.stringify({
```

Comment:
This has trouble being stringified for some reason. I was able to resolve this by doing a traditional key pair object instead.

See the relevant function for the comment.